lighthouse-core/gather/gatherers/dobetterweb/domnodes.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Gathers all DOM nodes on the page.
+ */
+
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+class AllDOMNodes extends Gatherer {
+  /**
+   * @param {!Object} options
+   * @return {!Promise<!Array<!Object>>}
+   */
+  afterPass(options) {
+    return options.driver.querySelectorAll('body, body /deep/ *');
```

Comment:
`DOM.getFlattenedDocument`? I did a quick dirty benchmark between the two and found `querySelectorAll` to be faster:

Runs: 1337ms, 1033ms
```
  console.time('took');
  return options.driver.sendCommand('DOM.enable')
    .then(_ => options.driver.sendCommand('DOM.getFlattenedDocument', {
      depth: -1, pierce: true
    }))
    .then(result => {
      const nodes = result.nodes.filter(node => node.nodeType === 1);
      console.timeEnd('took');
      return nodes;
    });
```
 
Runs: 812ms, 425ms

```
    console.time('took');
    return options.driver.querySelectorAll('html, html /deep/ *')
      .then(nodes => {
        console.timeEnd('took');
        return nodes;
      });
```

IMO, `DOM.getFlattenedDocument` is a bit clunky to use. You have to filter for element nodes, but the biggest drawback is that `depth: -1, pierce: true` also traverses into iframes. We definitely don't want that.